api/functions/updateCounts.js

```diff
@@ -1,32 +1,65 @@
 import logger from "../utils/logger.js";
 
 /**
- * Updates the counts object with the messages and reactions from the slack data
- * @param {Object} counts - The counts object
- * @param {Object} slackData - The slack data
- * @returns {Object} The updated counts object
+ * Processes messages to count how many messages, reactions, and reactions received each user has.
+ *
+ * @param {Array} messages - An array of message objects. Each message includes:
+ *   - `user`: The user who sent the message.
+ *   - `reactions`: An array of users who reacted to the message.
+ *
+ * @returns {Object} An object with each user's message count, reaction count, and reactions received.
  */
-export const updateCounts = ({ counts, slackData }) => {
-	logger.debug(counts, slackData);
-	// @todo Implement this function
-	return {
-		"01-02-2025": {
-			U1234: {
-				messages: 12,
-				reactions: 22,
-			},
-			U5678: {
-				messages: 3,
-			},
-		},
-		"02-02-2025": {
-			U1234: {
-				messages: 33,
-				reactions: 45,
-			},
-			U5678: {
-				reactions: 11,
-			},
-		},
-	};
+export const updateCounts = (messages) => {
+	const activity = {};
+
+	if (!messages || !Array.isArray(messages)) {
+		return {};
+	}
+
+	try {
+		messages.forEach((message) => {
+			if (!message || typeof message !== "object") {
+				return; // Skip this iteration
+			}
```

Comment:
```suggestion
		messages.filter(msg => msg && typeof msg === "object").forEach((message) => {
```

This is subjective and completely optional but you could also use `filter` here to achieve the same thing